packages/ui-components/src/components/Matrix/MatrixRow.tsx

```diff
@@ -194,10 +255,22 @@ export const MatrixRow = ({ row, parents = [] }: MatrixRowProps) => {
 
   const isCategory = children ? children.length > 0 : false;
 
-  // render a regular row with the title cell and the values
+  const getClassNames = () => {
+    const highlightedClass = isExpanded || depth > 0 ? 'highlighted' : '';
+    const matrixClass = 'matrix';
+    const childClass = depth > 0 ? 'child' : 'parent';
+
+    const baseClass = `${matrixClass} ${highlightedClass} ${childClass}`;
+    if (depth > 0 || index === undefined) return baseClass;
+    if (index % 2 === 0) return `${baseClass} even`;
+    return `${baseClass} odd`;
+  };
```

Comment:
Class names here seemed like a better way to handle things, and to be able to set colours in `tupaia-web` as well, as opposed to props